packages/shared/src/features/onboarding/steps/FunnelQuiz.tsx

```diff
@@ -0,0 +1,95 @@
+import type { ReactElement } from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
+import type { FunnelStepQuiz } from '../types/funnel';
+import {
+  FunnelStepQuizQuestionType,
+  FunnelStepTransitionType,
+} from '../types/funnel';
+import { FormInputRating } from '../../common/components/FormInputRating';
+import { FormInputCheckboxGroup } from '../../common/components/FormInputCheckboxGroup';
+import ConditionalWrapper from '../../../components/ConditionalWrapper';
+import { FunnelStepCtaWrapper } from '../shared/FunnelStepCtaWrapper';
+
+const quizComponentsMap = {
+  [FunnelStepQuizQuestionType.Rating]: FormInputRating,
+  [FunnelStepQuizQuestionType.Radio]: FormInputCheckboxGroup,
+  [FunnelStepQuizQuestionType.Checkbox]: FormInputCheckboxGroup,
+};
+
+const checkIfSingleChoice = (type: FunnelStepQuizQuestionType): boolean => {
+  return (
+    type === FunnelStepQuizQuestionType.Radio ||
+    type === FunnelStepQuizQuestionType.Rating
+  );
+};
+
+export const FunnelQuiz = ({
+  id,
+  question,
+  onTransition,
+}: FunnelStepQuiz): ReactElement => {
+  const { type, text, options, imageUrl } = question;
+  const isSingleChoice = checkIfSingleChoice(type);
+  const [stepValue, setStepValue] = useState<string | string[]>([]);
+  const Component = useMemo(() => quizComponentsMap[type], [type]);
+  const inputOptions = useMemo(
+    () =>
+      options.map((option) => ({
+        label: option.label,
+        value: option.label,
+      })),
+    [options],
+  );
+
+  const onChange = useCallback(
+    (input: string | string[]) => {
+      const value =
+        isSingleChoice && Array.isArray(input) ? input.at(-1) : input;
+      setStepValue(value);
+
+      if (isSingleChoice) {
+        onTransition?.({
+          type: FunnelStepTransitionType.Complete,
+          details: { value },
+        });
+      }
+    },
+    [isSingleChoice, onTransition],
+  );
+
+  const onCtaClick = useCallback(() => {
+    if (isSingleChoice || !stepValue.length) {
+      return;
+    }
+    onTransition?.({
+      type: FunnelStepTransitionType.Complete,
+      details: { value: stepValue },
+    });
+  }, [isSingleChoice, stepValue, onTransition]);
+
+  return (
+    <ConditionalWrapper
+      condition={!isSingleChoice}
```

Comment:
Just pointing it out that this wrapper will also be applied when it is `Rating`.

Reply:
Why do you think that? On storybook is not applied https://storybook-git-mi-849-funnel-form-inputs-dailydotdev.vercel.app/?path=/docs/components-onboarding-funnelquiz--docs